File: core/src/main/java/com/graqr/threshr/model/redsky/product/Rating.java
package com.graqr.threshr.model.redsky.product;

import com.fasterxml.jackson.annotation.JsonProperty;
import io.micronaut.core.annotation.Nullable;
import io.micronaut.serde.annotation.Serdeable;

import java.util.List;

@Serdeable
public record Rating(
        @Nullable Double average,
        @Nullable Long count,
        @Nullable Distribution distribution,
        @JsonProperty("secondary_averages")
        @Nullable List<SecondaryAverage> secondaryAverages,
        @Nullable Double value,
        @JsonProperty("submitted_at")
        String submittedAt) {
}

File: src/main/kotlin/com/graqr/threshr/circleStore/model/product/Rating.kt
package com.graqr.threshr.circleStore.model.product

import com.fasterxml.jackson.annotation.JsonProperty
import io.micronaut.core.annotation.Nullable
import io.micronaut.serde.annotation.Serdeable

@Serdeable
data class Rating(
    @param:Nullable val average: Double,
    @param:Nullable val count: Long,
    @param:Nullable val distribution: Distribution,
    @param:JsonProperty("secondary_averages") @param:Nullable val secondaryAverages: List<SecondaryAverage>,
    @param:Nullable val value: Double,
    @param:JsonProperty("submitted_at") val submittedAt: String
) {}
